### PR TITLE
community[patch]: Improve ChatBaichuan init args and role

### PR DESCRIPTION
**Pull request by maang-h**
- **Description:** Improve ChatBaichuan init args and role
   -  ChatBaichuan adds `system` role
   - alias: `baichuan_api_base` -> `base_url`
   - `with_search_enhance` is deprecated
   - Add `max_tokens` argument

### CODE REVIEW

File: libs/community/langchain_community/chat_models/baichuan.py [hunk 1]
```diff
@@ -113,14 +121,16 @@ def lc_secrets(self) -> Dict[str, str]:
...
-    baichuan_api_base: str = Field(default=DEFAULT_API_BASE)
+    baichuan_api_base: str = Field(default=DEFAULT_API_BASE, alias="base_url")
     """Baichuan custom endpoints"""
     baichuan_api_key: SecretStr = Field(alias="api_key")
     """Baichuan API Key"""
     baichuan_secret_key: Optional[SecretStr] = None
     """[DEPRECATED, keeping it for for backward compatibility] Baichuan Secret Key"""
     streaming: bool = False
     """Whether to stream the results or not."""
+    max_tokens: int = 2048
```
Comment (ccurme) [inline libs/community/langchain_community/chat_models/baichuan.py:132]: would this introduce a new token limit by default?

Comment (maang-h) [reply libs/community/langchain_community/chat_models/baichuan.py:132]: Yes, the default value for max tokens introduced in the latest documentation is 2048.

Comment (ccurme) [reply libs/community/langchain_community/chat_models/baichuan.py:132]: What is the benefit of maintaining that default here? If it changes Langchain will have an incorrect default. Does it make sense for this to be `Optional[int] = None`?

Comment (maang-h) [reply libs/community/langchain_community/chat_models/baichuan.py:132]: Wow! I understand, I will improve it.

**Review by ccurme — APPROVED** [review]

**Pull request merged**